src/convertToObject.js

```diff
@@ -6,7 +6,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const convertedElements = {};
+  const styleElements = sourceString.split(';');
+
+  for (const element of styleElements) {
+    const [property, value] = element.split(':');
+
+    if (value !== undefined) {
+      convertedElements[`${property.trim()}`] = `${value.trim()}`;
+    }
```

Comment:
[CODE PRACTICE] - Instead of using a for...of loop, consider using array iteration methods like `filter` to exclude undefined values and `map` to transform elements into the desired shape. This approach aligns better with functional programming practices and the checklist item that advises avoiding loops in this task.